### PR TITLE
feat(android-setup): wait for port forwarding removal before application exits

### PR DESCRIPTION
**Pull request by karanbirsingh**
#### Description of changes

This fixes the issue outlined in #3000 - before this PR, running `npx cross-env ANDROID_HOME=drop/mock-adb electron drop/electron/unified-dev/product/bundle/main.bundle.js` twice will lead to port-forwarding-failure the second time (See #3000 for debugging commands).

After this PR the application can be run multiple times and should clean up the port each time robustly.

#### Pull request checklist

- [ ] Addresses an existing issue: #0000
- [ ] Ran `yarn fastpass`
- [ ] Added/updated relevant unit test(s) (and ran `yarn test`)
- [ ] Verified code coverage for the changes made. Check coverage report at: `<rootDir>/test-results/unit/coverage`
- [ ] PR title *AND* final merge commit title both start with a semantic tag (`fix:`, `chore:`, `feat(feature-name):`, `refactor:`). See `CONTRIBUTING.md`.
- [ ] (UI changes only) Added screenshots/GIFs to description above
- [ ] (UI changes only) Verified usability with NVDA/JAWS

### CODE REVIEW

File: src/electron/platform/android/setup/android-port-cleaner.ts [hunk 1]
```diff
@@ -37,7 +37,6 @@ export class AndroidPortCleaner {
                 if (p) {
                     try {
                         await this.serviceConfig.removeTcpForwarding(p);
-                        await this.serviceConfig.listForwardedPorts(); // Allows ADB to complete
```
Comment (JGibson2019) [inline src/electron/platform/android/setup/android-port-cleaner.ts:40]: Why did you remove the call to list the forwarded ports here?

File: src/electron/ipc/ipc-event-action.ts [hunk 2]
```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+export class IpcEventAction {
```
Comment (DaveTryon) [inline src/electron/ipc/ipc-event-action.ts:3]: This should have unit tests, but the rest looks good to me

File: src/electron/ipc/ipc-event-action.ts [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+    private listeners: (() => Promise<void>)[] = [];
+
+    public async invokeAsync(): Promise<void> {
```
Comment (DaveTryon) [inline src/electron/ipc/ipc-event-action.ts:6]: This may need to be an arrow function to get the correct `this`. Unit tests will make that clear

File: src/electron/platform/android/setup/android-port-cleaner.ts (hunk 1 above)
Comment (karanbirsingh) [reply src/electron/platform/android/setup/android-port-cleaner.ts:40]: good question - the comment & prior PR show it was intended to try helping ADB commands fully complete before the app closed, but it turns out this behavior was due to our missing async listeners & not because of appium itself; we don't need the list for the production flow

File: src/electron/ipc/ipc-event-action.ts (hunk 3 above)
Comment (karanbirsingh) [reply src/electron/ipc/ipc-event-action.ts:6]: I think it is OK, I've compared it to the methods I see in [Action](https://github.com/microsoft/accessibility-insights-web/blob/master/src/common/flux/action.ts), can you let me know if I'm missing something in particular?

File: src/electron/ipc/ipc-event-action.ts (hunk 2 above)
Comment (dbjorge) [inline src/electron/ipc/ipc-event-action.ts:3]: I think this approach seems reasonable, but I think it would make more sense to call this an `AsyncAction`; I don't think what it's actually doing is really related to `IpcEvent`s except insofar as that happens to be how it's consumed

File: src/electron/ipc/ipc-event-action.ts (hunk 3 above)
Comment (DaveTryon) [reply src/electron/ipc/ipc-event-action.ts:6]: If unit tests are happy and it enables the E2E tests, I'm good :)

File: src/tests/unit/tests/electron/ipc/async-action.test.ts [hunk 4]
```diff
@@ -0,0 +1,20 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+import { AsyncAction } from 'electron/ipc/async-action';
+import { Mock, Times } from 'typemoq';
+
+describe('IpcEventAction', () => {
```
Comment (karanbirsingh) [inline src/tests/unit/tests/electron/ipc/async-action.test.ts:6]:
```suggestion
describe('AsyncAction', () => {
```
